src/api/notify-namespace.ts

```diff
@@ -0,0 +1,538 @@
+        throw new Error(
+          'Nft Activity Webhooks require a non-empty array input.'
+        );
+      }
+      network = params.network
+        ? NETWORK_TO_WEBHOOK_NETWORK.get(params.network)
+        : network;
+      filters = (params.filters as NftFilter[]).map(filter => ({
+        contract_address: filter.contractAddress,
+        token_id: filter.tokenId
+      }));
+    } else if (type === WebhookType.ADDRESS_ACTIVITY) {
+      if (
+        params === undefined ||
+        !('addresses' in params) ||
+        params.addresses.length === 0
+      ) {
+        throw new Error(
+          'Address Activity Webhooks require a non-empty array input.'
+        );
+      }
+      network = params.network
+        ? NETWORK_TO_WEBHOOK_NETWORK.get(params.network)
+        : network;
+      addresses = params.addresses;
+    }
+
+    const data = {
+      network,
+      webhook_type: type,
+      webhook_url: url,
+      ...(appId && { app_id: appId }),
+
+      // Only include the filters/addresses in the final response if it's defined
+      ...(filters && { nft_filters: filters }),
+      ...(addresses && { addresses })
+    };
+
+    const response = await requestHttpWithBackoff<{}, RawCreateWebhookResponse>(
+      this.config,
+      AlchemyApiType.WEBHOOK,
+      'create-webhook',
+      'webhooks:createWebhook',
+      {},
+      {
+        headers: {
+          'X-Alchemy-Token': this.config.notifyAuthToken!
+        },
+        method: 'POST',
+        data
+      }
+    );
+
+    return parseRawWebhook(response.data);
+  }
+
+  /**
+   * Delete the provided webhook.
+   *
+   * @param webhook The webhook to delete.
+   */
+  deleteWebhook(webhook: Webhook): Promise<void>;
+
+  /**
+   * Delete the provided webhook.
+   *
+   * @param webhookId The id of the webhook to delete.
+   */
+  deleteWebhook(webhookId: string): Promise<void>;
+  async deleteWebhook(webhookOrId: Webhook | string): Promise<void> {
+    this.verifyConfig();
+    const webhookId =
+      typeof webhookOrId === 'string' ? webhookOrId : webhookOrId.id;
+    const response = await requestHttpWithBackoff<{}, RawNftFiltersResponse>(
+      this.config,
+      AlchemyApiType.WEBHOOK,
+      'delete-webhook',
+      'webhooks:deleteWebhook',
+      {
+        webhook_id: webhookId
+      },
+      {
+        headers: { 'X-Alchemy-Token': this.config.notifyAuthToken! },
+        method: 'DELETE'
+      }
+    );
+
+    if ('message' in response) {
+      throw new Error(
+        `Webhook not found. Failed to delete webhook: ${webhookId}`
+      );
+    }
+  }
+
+  private verifyConfig() {
+    if (this.config.notifyAuthToken === undefined) {
+      throw new Error(
+        'Using the Notify API requires setting the Alchemy Auth Token in ' +
+          'the settings object when initializing Alchemy.'
+      );
+    }
+  }
+}
+
+/**
+ * Mapping of webhook network representations to the SDK's network representation.
+ *
+ * @internal
+ */
```

Comment:
Maybe make this clear this is just supported Webhook networks and doesn't include unsupported networks like Solana / Astar yet? 

Unless it's a 1:1 mapping to the SDK Network object, in which case it would probably make more sense to just skip the mapping and use the original object

Reply:
Good point, added a sentence at the top of `NotifyNamespace`. It's not a 1:1 mapping, but we need the mappings here internally to convert the SDK's `Network` num into the string format that the Notify API requires (and vice versa when creating the `Webhook` object).